### ISSUE TITLE
Problem with setting gamma when defining an Image layer with `viewer.add_image`

### ISSUE THREAD

**Issue by jules-vanaret** [napari/napari#7005]
### 🐛 Bug Report

When setting the gamma value of an image layer in `viewer.add_image(..., gamma=...)`, the gamma value does not get applied once the window is displayed. Clicking on the same gamma value on the gamma slider in the layer interface updates the image with the correct gamma.

Here's a video produced using the reproducer below. I display to image layer with the same data but different gamma values. The both show with the same appearance. After I click on the correct gamma value of the gamma slider of the problematic layer (choosing exactly the same gamma value as specified initially), the layer display updates and show the correct appearance.

https://github.com/napari/napari/assets/94963445/9ec08156-e1f9-4d7b-8e89-0b578f68583d

### 💡 Steps to Reproduce

You can use the following reproducer:
```python
import numpy as np
import napari


_,img= np.meshgrid(np.linspace(0, 1, 256), np.linspace(0, 1, 256))

viewer = napari.Viewer()

viewer.add_image(img, colormap='reds', gamma=1.5, translate=[0,10])
viewer.add_image(img, colormap='reds')

viewer.grid.enabled = True
viewer.grid.shape = (1, 2) 

napari.run()
```

### 💡 Expected Behavior

An image layer should display with the gamma value specified when calling `add_image`. 

### 🌎 Environment

napari: 0.4.19.post1
Platform: Linux-5.15.0-105-generic-x86_64-with-glibc2.31
System: Ubuntu 20.04.6 LTS
Python: 3.10.14 | packaged by conda-forge | (main, Mar 20 2024, 12:45:18) [GCC 12.3.0]
Qt: 5.15.2
PyQt5: 5.15.10
NumPy: 1.26.4
SciPy: 1.13.1
Dask: 2024.6.0
VisPy: 0.14.3
magicgui: 0.8.3
superqt: 0.6.7
in-n-out: 0.2.1
app-model: 0.2.7
npe2: 0.7.5

OpenGL:
  - GL version:  4.6 (Compatibility Profile) Mesa 21.2.6
  - MAX_TEXTURE_SIZE: 16384

Screens:
  - screen 1: resolution 1920x1080, scale 1.0
  - screen 2: resolution 2560x1440, scale 1.0

Settings path:
  - ~/.config/napari/napari-test_a73bc024590c6d21bb1fcfadf2850488aaa4bbba/settings.yaml
Plugins:
  - napari: 0.4.19.post1 (77 contributions)
  - napari-console: 0.0.9 (0 contributions)
  - napari-svg: 0.1.10 (2 contributions)

**Comment by Czaki** [comment]
it looks like the problem was resolved in #6650 and will be fixed in 0.5.0 release

**Comment by jules-vanaret** [comment]
Cool, thank you very much for the PR.
